src/test/java/org/jabref/logic/importer/fetcher/SpringerFetcherTest.java

```diff
@@ -182,9 +182,9 @@ public void supportsBooleanANDSearch() throws Exception {
 
         Assertions.assertTrue(resultJustByAuthor.containsAll(result));
         List<BibEntry> allEntriesFromCSCW = result.stream()
-                                                  .filter(bibEntry -> bibEntry.getField(StandardField.JOURNAL)
-                                                                              .orElse("")
-                                                                              .equals("Computer Supported Cooperative Work (CSCW)"))
+                                                  .filter(bibEntry -> "Computer Supported Cooperative Work (CSCW)"
+                                                                              .equals(bibEntry.getField(StandardField.JOURNAL)
+                                                                              .orElse("")))
```

Comment:
This seems to be NPE safe before, as the Optional returns either the contents of the field or an empty string. In either case equals wont be run on null.

Reply:
These are the rules 🤷‍♂️. Should we disable the rule and only commit the changes manually? 😅

Reply:
I think this is an edge case here, but in general the rule is useful

Reply:
So there seems to be a problem with the rules, unable to understand optionals. Let's talk in the devcall about pros and cons

Reply:
I filed https://github.com/openrewrite/rewrite-static-analysis/issues/97 - think, we should close this pull request.

I also find the rewrites not intuitive. Let's discuss.